### ISSUE TITLE
Zarf tools archiver fails to overwrite files

### ISSUE THREAD

**Issue by runyontr** [defenseunicorns/zarf#1339]
When bundling files and binaries into zarf packages as part of the #1144 change, the user of the command `zarf tools archiver` is shown as the default way to extract binaries from tarballs.  e.g. [here](https://github.com/defenseunicorns/zarf/blob/main/examples/terraform/zarf.yaml#L13).     When performing an extraction to the same location, this tool errors out.  

### Steps to reproduce
1. Take a zarf package and extract the file to a location:

```yaml
kind: ZarfPackageConfig
metadata:
  name: "eksctl"
  description: "Run terraform/terragrunt code"

components:
  - name: download-eksctl
    required: true
    actions:
      onDeploy:
        after:
          - cmd: zarf tools archiver decompress tmp/eksctl_Darwin_arm64.tar.gz .
          - cmd: rm -rf tmp
    files:
      # mac m1 eksctl binary
      - source: https://github.com/weaveworks/eksctl/releases/download/v0.128.0/eksctl_Darwin_arm64.tar.gz
        target: tmp/eksctl_Darwin_arm64.tar.gz

  - name: execute-eksctl
    actions:
      onDeploy:
        before:
          - cmd: "./eksctl create cluster -f cluster.yaml"
      onRemove:
        before:
          - cmd: "./eksctl delete cluster -f cluster.yaml --disable-nodegroup-eviction --wait"
```

```bash
$ zarf package create
$ zarf package deploy zarf-package-eksctl-arm64.tar.zst --confirm # succeeds
$ zarf package deploy zarf-package-eksctl-arm64.tar.zst --confirm
...
  ⠋  Waiting for command "zarf tools archiver decompress tmp/eksctl_Darwin_arm64.tar.gz ." (no timeout) (0s)                                                                                              
   sh [-c zarf tools archiver decompress tmp/eksctl_Darwin_arm64.tar.gz .]
     ERROR:  Unable to perform decompression
  ✔  Running command "zarf tools archiver decompress tmp/eksctl_Darwin_arm64.tar.gz ."                                                                                                                    
     ERROR:  Failed to deploy package: unable to deploy all components in this Zarf Package: unable to deploy
             component download-eksctl: unable to run component after action: command "zarf tools archiver
             decompress tmp/eksctl_Darwin_arm64.tar.gz ." failed after 0 retries
$ rm eksctl
$ zarf package deploy zarf-package-eksctl-arm64.tar.zst --confirm # succeeds
```

**Comment by UncleGedd** [comment]
FYSA, I reported something similar to this issue a while ago: https://github.com/defenseunicorns/zarf/issues/858

tldr: need to update `mholt/archiver` but there are breaking changes

**Comment by Noxsios** [comment]
The functionality needed here is mostly solved by #1928, closing unless there is continued need.
